### PR TITLE
Add policy result representation

### LINKED ISSUE #384: Add PolicyResultRepresentation
### Description

https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/core/src/main/java/org/keycloak/representations/idm/authorization/PolicyEvaluationResponse.java#L119

### Motivation

This type is necessary in order to access a resource evaluation result in Clients -> Authorization -> Evaluate

### PR DESCRIPTION
**Pull request by jenny-s51**
Closes #384

### CODE REVIEW

File: src/defs/policyRepresentation.ts [hunk 1]
```diff
@@ -8,6 +8,11 @@ export enum DecisionStrategy {
   CONSENSUS = 'CONSENSUS',
 }
 
+export enum DecisionEffect {
+  PERMIT = 'PERMIT',
+  DENY = 'DENY',
```
Comment (jonkoops) [inline src/defs/policyRepresentation.ts:13]: Let's use the same conventions here as in the [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/enums.html), which is to camel-case the name of the key.
```suggestion
  Permit = 'PERMIT',
  Deny = 'DENY',
```

File: src/defs/policyResultRepresentation.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import PolicyRepresentation, {DecisionEffect} from './policyRepresentation';
+import ScopeRepresentation from './scopeRepresentation';
+
+export default interface PolicyResultRepresentation {
+  policy: PolicyRepresentation;
+  status: DecisionEffect;
+  associatedPolicies: PolicyResultRepresentation[];
+  scopes: ScopeRepresentation[];
```
Comment (jonkoops) [inline src/defs/policyResultRepresentation.ts:8]: @jenny-s51 Do you have a link to the documentation for this type? I cannot seem to find it in the documentation.

I think we also have to mark these fields as optional, as the API does not guarantee they are actually set (which is a nice think we inherit from the design flaws of Java).

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/defs/policyResultRepresentation.ts (hunk 2 above)
Comment (jenny-s51) [reply src/defs/policyResultRepresentation.ts:8]: The type doesn't seem to exist in the docs and can be found here @jonkoops : https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/core/src/main/java/org/keycloak/representations/idm/authorization/PolicyEvaluationResponse.java#L119-L124

File: src/defs/policyResultRepresentation.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import PolicyRepresentation, {DecisionEffect} from './policyRepresentation';
+import ScopeRepresentation from './scopeRepresentation';
+
+export default interface PolicyResultRepresentation {
+  policy?: PolicyRepresentation;
+  status?: DecisionEffect;
+  associatedPolicies?: PolicyResultRepresentation[];
+  scopes?: ScopeRepresentation[];
```
Comment (jonkoops) [inline src/defs/policyResultRepresentation.ts:8]: According to the code you linked this should be `string[]`, or is there some transformation going on to link the full `ScopeRepresentation`? See the source code [here](https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/core/src/main/java/org/keycloak/representations/idm/authorization/PolicyEvaluationResponse.java#L167).

**Review by jonkoops — CHANGES_REQUESTED** [review]
Possibly one small type mismatch, but this looks good to me otherwise.

File: src/defs/policyResultRepresentation.ts (hunk 2 above)
Comment (jonkoops) [reply src/defs/policyResultRepresentation.ts]: Great, thanks for linking!

File: src/defs/policyResultRepresentation.ts (hunk 3 above)
Comment (jenny-s51) [reply src/defs/policyResultRepresentation.ts]: Ah, nice catch! Yes - fixed.

**Review by jonkoops — APPROVED** [review]
LGTM!

**Pull request merged**